train_ta.py

```diff
@@ -0,0 +1,330 @@
+arg_parser.add_argument("--save_model_threshold", type=float, default=1.0)
+arg_parser.add_argument("--gp_loss_weight", type=float, default=0.01)
+arg_parser.add_argument("--gp_inference_index", type=int, default=10)
+arg_parser.add_argument("--disable_wandb", dest="enable_wandb", action="store_false")
+arg_parser.set_defaults(enable_wandb=True)
+
+config = arg_parser.parse_args().__dict__
+
+# Random seeds
+assert config["satellite"] in ["sentinel_2", "planet_daily"]
+assert config["pos"] in ["both", "34S_19E_258N", "34S_19E_259N"]
+assert config["split_by"] in [None, "latitude", "longitude"]
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Data loaders
+# ----------------------------------------------------------------------------------------------------------------------
+kwargs = dict(
+    satellite=config["satellite"],
+    include_bands=True,
+    include_cloud=True,
+    include_ndvi=False,
+    include_rvi=False,
+    image_size=None,
+    spatial_backbone="random_pixel",
+    min_area_to_ignore=1000,
+    train_or_test="train",
+    temporal_augmentation=False,
+)
+if config["pos"] == "both":
+    label_names_258, reader_258 = load_reader(pos="34S_19E_258N", **kwargs)
+    print("\u2713 Loaded 258")
+    label_names_259, reader_259 = load_reader(pos="34S_19E_259N", **kwargs)
+    print("\u2713 Loaded 259")
+    assert (
+        label_names_258 == label_names_259
+    ), f"{label_names_258} and {label_names_259} are not equal"
+    label_names = label_names_258
+    reader = torch.utils.data.ConcatDataset([reader_258, reader_259])
+    reader.labels = pd.concat([reader_258.labels, reader_259.labels], ignore_index=True)
+else:
+    label_names, reader = load_reader(pos=config["pos"], **kwargs)
+
+
+config["input_dim"] = reader[0][0].shape[1]
+config["sequence_length"] = reader[0][0].shape[0]
+config["X_shape"] = reader[0][0].shape
+config["output_timesteps"] = config["sequence_length"] - config["input_timesteps"]
+
+print("\u2713 Datasets initialized")
+
+data_loader = DataLoader(
+    train_val_reader=reader,
+    validation_split=config["validation_split"],
+    split_by=config["split_by"],
+)
+train_loader = data_loader.get_train_loader(batch_size=config["batch_size"], num_workers=0)
+valid_loader = data_loader.get_validation_loader(batch_size=config["batch_size"], num_workers=0)
+
+config["train_dataset_size"] = len(train_loader.dataset)
+config["train_minibatch_size"] = len(train_loader)
+config["val_dataset_size"] = len(valid_loader.dataset)
+config["val_minibatch_size"] = len(valid_loader)
+
+print("\u2713 Data loaders initialized")
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Initialize model
+# ----------------------------------------------------------------------------------------------------------------------
+class TemporalAugmentor(nn.Module):
```

Comment:
This is the new class you created, correct?

Reply:
yes